response_operations_ui/templates/edit-contact-details.html

```diff
@@ -0,0 +1,124 @@
+                {{ form.csrf_token }}
+                {% if error %}
+                <div class="panel panel--simple panel--error u-mb-l">
+                  <div class="panel__body" id="save-error">
+                    <h2 id='error-saving-message' class="venus">Error saving changes</h2>
+                    <p class="mars">Please try again.</p>
+                  </div>
+                </div>
+
+                {% endif %}
+
+                <div class="field u-mb-l">
+                    <label class="label" for="lastName">
+                      Last name
+                      <span class="secondary-info current-value">
+                        <span class="u-vh">Current value: </span>
+                        {{ respondent_details.lastName }}
+                      </span>
+                    </label>
+                  {{ form.last_name(class_="input input--text", id='lastName', required=required, maxlength=254) }}
+                </div>
+
+                <div class="field u-mb-l">
+                    <label class="label" for="firstName">
+                      First name
+                      <span class="secondary-info current-value">
+                        <span class="u-vh">Current value: </span>
+                        {{ respondent_details.firstName }}
+                      </span>
+                    </label>
+                  {{ form.first_name(class_="input input--text", id='firstName', required=required, maxlength=254) }}
+                </div>
+
+                <div class="field u-mb-l">
+                    <label class="label" for="email">
+                      Email address
+                      <span class="secondary-info current-value">
+                        <span class="u-vh">Current value: </span>
+                        {{ respondent_details.emailAddress }}
+                      </span>
+                      <span class="label__description pluto">ONS will send a verification email to a new address</span>
+                    </label>
+                  {{ form.email(class_="input input--text", id='email', required=required, maxlength=254, type="email") }}
+                </div>
+
+                <div class="field u-mb-m">
+                    <label class="label" for="telephone">
+                      Phone number
+                      <span class="secondary-info current-value">
+                        <span class="u-vh">Current value: </span>
+                        {{ respondent_details.telephone }}
+                      </span>
+                    </label>
+                  {{ form.telephone(class_="input input--text", id='telephone', required=required, maxlength=254) }}
+                </div>
+
+                <button class="btn btn--primary u-mt-s" id="save-btn">Save changes</button>
+                </form>
+
+                <div>
+                  <a id="cancel-btn" href="{{ referrer }}">Cancel</a>
```

Comment:
Could we please change the href URL to `"{{ url_for('reporting_unit_bp.view_reporting_unit', ru_ref=ru_ref) }}"`? We should not use `referrer` here because if an internal user is navigating to the `Edit contact details` page after successfully editing another respondent's details, the referrer would be set as `/reporting-units/<ru_ref>?edit_details=True`. In this case, the internal user will see `Contact details changed` message even after selecting the `cancel-btn`.